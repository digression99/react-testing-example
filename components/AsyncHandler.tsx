import { useState } from 'react'
import { Box, Button, Text} from '@chakra-ui/react'

export const AsyncHandler = () => {
  const [isShown, setIsShown] = useState(false)

  const onClick = () => {
    if (isShown) return setIsShown(false)
    setTimeout(() => {
      setIsShown(true)
    }, 500)
  }

  return (
    <Box>
      <Button onClick={onClick}>
        Show the text after 1 seconds
      </Button>
      {isShown && (
        <Text>Text appeared!</Text>
      )}
    </Box>
  )
}
