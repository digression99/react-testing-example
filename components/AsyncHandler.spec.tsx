import userEvent from '@testing-library/user-event'
import { render, screen } from '@testing-library/react'
import { AsyncHandler } from "./AsyncHandler";

describe("<AsyncComponent />", () => {

  beforeEach(() => {
    jest.useFakeTimers()
  })

  test('renders correctly', () => {
    const { container } = render(<AsyncHandler />)
    expect(container).toMatchSnapshot()
  })

  test('click the button', async () => {
    const user = userEvent.setup()
    render(<AsyncHandler />)
    const button = screen.getByRole('button', { name: /Show the text/i })
    expect(button).toBeInTheDocument()
    user.click(button)
    const text = await screen.findByText(/appeared/i, {}, { timeout: 2000 })
    expect(text).toBeInTheDocument()
  })
})