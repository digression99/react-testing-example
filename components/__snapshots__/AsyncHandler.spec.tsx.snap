// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AsyncComponent /> renders correctly 1`] = `
<div>
  <div
    class="css-0"
  >
    <button
      class="chakra-button css-0"
      type="button"
    >
      Show the text after 1 seconds
    </button>
  </div>
</div>
`;
